client/SFTPClientGUI/SFTPClientGUI.py

```diff
@@ -0,0 +1,351 @@
+		self.upload_button = upload_button
+		self.load_dirs(os.path.abspath(os.sep))
+		self.treeview_local.set_model(self._tvmodel)
+
+	def load_dirs(self, path, parent=None):
+		counter = 0
+		for name in os.listdir(path):
+			if self.local_hidden and name.startswith('.'):
+				continue
+			fullname = os.path.join(path, name)
+			r_permissions = os.access(fullname, os.R_OK)
+			w_permissions = os.access(fullname, os.W_OK)
+			x_permissions = os.access(fullname, os.X_OK)
+			perm = '   '
+			perm = perm + 'r' if r_permissions else perm + '-'
+			perm = perm + 'w' if w_permissions else perm + '-'
+			perm = perm + 'x' if x_permissions else perm + '-'
+			date_modified = '   ' + time.strftime("%y-%m-%d %H:%M:%S", time.localtime(os.path.getmtime(fullname)))
+			isFolder = stat.S_ISDIR(os.stat(fullname).st_mode)
+			if isFolder:
+				icon = Gtk.IconTheme.get_default().load_icon('folder', 20, 0)
+				if not (r_permissions and x_permissions):
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, None, -1, date_modified))
+			else:
+				file_size = os.path.getsize(fullname)
+				hr_file_size = '   ' + boltons.strutils.bytes2human(file_size)
+				if not (r_permissions and x_permissions):
+					icon = Gtk.IconTheme.get_default().load_icon('emblem-unreadable', 13, 0)
+				icon = Gtk.IconTheme.get_default().load_icon('empty', 12.5, 0)
+				current = self._tvmodel.append(parent, (name, icon, fullname, perm, hr_file_size, file_size, date_modified))
+			if isFolder and (r_permissions and x_permissions):
+				self._tvmodel.append(current, [None, None, None, None, None, None, None])
+			counter += 1
+		if counter < 1:
+			self._tvmodel.append(parent, [None, None, None, None, None, None, None])
+
+	def delete(self, model, treeiter):
+		try:
+			if model[treeiter][5] == -1:
+				shutil.rmtree(model[treeiter][2])
+			else:
+				os.remove(model[treeiter][2])
+			self.logger.log_generic('Successfully deleted ' + model[treeiter][2])
+			self.refresh()
+		except OSError:
+			self.logger.log_generic("Permissions Error Deleting File")
+
+class RemoteDirectory(DirectoryBase):
+	def __init__(self, treeview, logger, download_button, application, ftp, *args, **kwargs):
+		super(RemoteDirectory, self).__init__(treeview, logger, *args, **kwargs)
+		self.ftp = ftp
+		self.download_button = download_button
+		self.application = application
+		self.def_dir = self.application.config['server_config']['server.web_root']
+		self.load_dirs(self.def_dir)
+		self.treeview_local.set_model(self._tvmodel)
+
+	def load_dirs(self, path, parent=None):
+		self.ftp.chdir(path)
+		counter = 0
+		for name in self.ftp.listdir(path):
+			if self.local_hidden and name.startswith('.'):
+				continue
+			fullname = path + '/' + name
+			lstatout = self.ftp.lstat(fullname)
+			perm = '   '
+			_file = self.ftp.file(fullname)
+			perm += 'r' if _file.readable() else '-'
```

Comment:
If you can call `stat` on the file, which I believe paramiko can do, you can get the execute permission as well and just use the bit-wise mask.